config/tf_modules/aws-cloudfront-distribution/main.tf

```diff
@@ -0,0 +1,92 @@
+resource "random_string" "default-suffix" {
+  length  = 7
+  special = false
+}
+
+data "aws_s3_bucket" "current_bucket" {
+  bucket = var.bucket_name
+}
+
+resource "aws_s3_bucket_object" "default_page" {
+  bucket       = data.aws_s3_bucket.current_bucket.id
+  key          = "opta-default-page-${random_string.default-suffix.result}.html"
+  source       = "${path.module}/default.html"
+  content_type = "text/html"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
```

Comment:
This resource and the one above it should not be created unless `var.bucket_name` is true. The cloudfront module will basically be useless without linking. But we should follow this to keep the module generic.